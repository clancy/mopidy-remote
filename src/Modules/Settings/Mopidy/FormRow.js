import React from 'react'
import {
  View,
  Text,
  TextInput,
  StyleSheet
} from 'react-native';

const FormRow = (props) => (
  <View style={styles.settingRow}>
    <Text style={styles.text}>{props.label}</Text>
    <TextInput
      style={styles.textInput}
      onChangeText={props.onChangeText}
      value={props.value} />
  </View>
);

const styles = StyleSheet.create({
  settingRow: {
    height: 40,
    flexDirection: 'row',
    alignItems: 'center',
    margin: 5
  },
  text: {
    color: 'white',
    width: 80
  },
  textInput: {
    color: 'white',
    borderColor: 'white',
    borderWidth: 1,
    flex: 1
  }
})

export default FormRow;
