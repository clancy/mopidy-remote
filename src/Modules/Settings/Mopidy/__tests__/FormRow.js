import 'react-native';
import React from 'react';
import FormRow from '../FormRow';
import { shallow, mount, render } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
describe('FormRow tests', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <FormRow
        label={ "label" }
        value={ "value" }
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('connects next control correctly', () => {
    var controlMock = jest.fn();
    const formRow = shallow(
      <FormRow
        label={ "label" }
        value={ "value" }
        onChangeText={() => {controlMock('onChangeText')}}
        />
    );
    formRow.findWhere(s => s.name() == 'TextInput').simulate('ChangeText', 'wenger');

    expect(controlMock.mock.calls.length).toBe(1);
    expect(controlMock.mock.calls[0][0]).toBe('onChangeText');
  });
});
