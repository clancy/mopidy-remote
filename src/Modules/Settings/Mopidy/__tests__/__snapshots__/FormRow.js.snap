// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRow tests renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 40,
      "margin": 5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "width": 80,
      }
    }
  >
    label
  </Text>
  <TextInput
    onChangeText={undefined}
    style={
      Object {
        "borderColor": "white",
        "borderWidth": 1,
        "color": "white",
        "flex": 1,
      }
    }
    value="value"
  />
</View>
`;
